samza-core/src/main/scala/org/apache/samza/storage/TaskStorageManager.scala

```diff
@@ -152,10 +151,7 @@ class TaskStorageManager(
       systemAdmin.validateStream(changelogSpec)
```

Comment:
This check appears to be redundant, since there is already a similar check at the AM. 
If we get rid of this check 
and modify getStartingOffset() method in this class to use sspMetadataCache.getMetadata(ssp).getOldestOffset

then we can remove this class's dependency on systemAdmins aswell (since it already has sspMetadataCache).

systemAdmins is not thread-safe, so this would help in parallelizing task-restores. 
Or I can send a separate PR with that change.